spikeinterface/preprocessing/highpass_spatial_filter.py

```diff
@@ -0,0 +1,280 @@
+import numpy as np
+import scipy.signal
+from spikeinterface.preprocessing.basepreprocessor import BasePreprocessor, BasePreprocessorSegment
+from spikeinterface.core.core_tools import define_function_from_class
+
+
+class HighpassSpatialFilter(BasePreprocessor):
+    """
+    Perform destriping with high-pass spatial filtering. Uses
+    the kfilt() function of the International Brain Laboratory.
+
+    Median average filtering, by removing the median of signal across
+    channels, assumes noise is constant across all channels. However,
+    noise have exhibit low-frequency changes across nearby channels.
+
+    This function extends the cut-band from 0 to 0.01 Nyquist to
+    remove these low-frequency changes during destriping.
+
+    Parameters
+    ----------
+
+    n_channel_pad: Number of channels to pad prior to filtering. Channels
+                   are padded with mirroring.
+
+    n_channel_taper: Number of channels to perform cosine tapering on
+                     prior to filtering. If None and n_channel_pad is set,
+                     n_channel_taper will be set to the number of
+                     padded channels. Otherwise, the passed value will be used.
+
+    agc_options: Options for automatic gain control. By default, gain control
+                 is applied prior to filtering to improve filter performance.
+                 "default" will use the IBL pipeline defaults (agc on). Setting to
+                 None will turn off gain control. To customise, pass a
+                 dictionary with the fields:
+                    agc_options["window_length_s"] - window length in seconds
+                    agc_options["sampling_interval"] - recording sampling interval
+
+    butter_kwargs: Dictionary with fields "N", "Wn", "btype" to be passed to
+                   scipy.signal.butter
+    """
+    name = 'highpass_spatial_filter'
+
+    def __init__(self, recording, n_channel_pad=None, n_channel_taper=None, agc_options="default", butter_kwargs=None):
+        BasePreprocessor.__init__(self, recording)
+
+        n_channel_pad, agc_options, butter_kwargs = self.handle_args(recording,
```

Comment:
We should check another important aspect: spatial kfilt assumes that channels are sorted by depth, but this is not always the case, so we should add a check and raise an error / warning if they are not sorted